### PR TITLE
feat: implement `useScrollbarWidth` utility hook

### PR DESCRIPTION
**Pull request by ling1726**
New hook that returns the width of the scrollbar in pixels of the current user agent

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/Avatar-mount_base.html">1319</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/Avatar-mount.html">1325</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/Button-mount_base.html">941</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/Button-mount.html">938</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/FluentProvider-mount_base.html">1593</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/FluentProvider-mount.html">1608</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/FluentProviderWithTheme-mount_base.html">628</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/FluentProviderWithTheme-mount.html">634</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">586</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">597</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">623</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">612</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/MakeStyles-mount_base.html">1892</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/MakeStyles-mount.html">1894</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/Persona-mount_base.html">3016</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/Persona-mount.html">3003</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/SpinButton-mount_base.html">2525</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26144/perf-test-react-components/SpinButton-mount.html">2562</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContext.fixture.js'>createContext</abbr>                                                                    |          `533 B`<br />`341 B` |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContextSelector.fixture.js'>createContextSelector</abbr>                                                    |          `554 B`<br />`348 B` |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr>                                           |  `80.097 kB`<br />`24.712 kB` |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `84.612 kB`<br />`21.445 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `51.515 kB`<br />`14.449 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |    `15.42 kB`<br />`6.167 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |  `67.733 kB`<br />`18.746 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                                                                       |    `23.78 kB`<br />`7.542 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                                                                         |    `24.67 kB`<br />`7.825 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                                                                       |   `24.437 kB`<br />`7.189 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |   `33.354 kB`<br />`8.779 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |  `40.384 kB`<br />`10.127 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |  `38.043 kB`<br />`10.096 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |  `45.488 kB`<br />`11.501 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |  `49.184 kB`<br />`10.875 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                                                |  `77.975 kB`<br />`22.496 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                                                          |   `72.92 kB`<br />`21.108 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                                              |     `9.04 kB`<br />`3.807 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                                              |   `10.805 kB`<br />`4.443 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                                            |    `9.847 kB`<br />`4.159 kB` |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr>                                                                              |   `29.081 kB`<br />`9.258 kB` |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/CheckboxField.fixture.js'>CheckboxField</abbr>                                                                    |   `35.51 kB`<br />`11.026 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |  `79.959 kB`<br />`25.572 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/ComboboxField.fixture.js'>ComboboxField</abbr>                                                                    |  `77.155 kB`<br />`25.536 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `79.205 kB`<br />`25.496 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |   `59.812 kB`<br />`16.62 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `188.779 kB`<br />`53.034 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.276 kB`<br />`11.309 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `83.459 kB`<br />`24.913 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                                                                 |   `16.938 kB`<br />`6.103 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                                                                       |   `11.259 kB`<br />`4.473 kB` |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoButton.fixture.js'>InfoButton</abbr>                                                                        | `119.624 kB`<br />`35.891 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `24.617 kB`<br />`8.059 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/InputField.fixture.js'>InputField</abbr>                                                                             |   `33.839 kB`<br />`10.73 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                                                                       |    `9.817 kB`<br />`4.067 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                                                                         |   `12.285 kB`<br />`5.063 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `119.492 kB`<br />`36.913 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `122.561 kB`<br />`37.436 kB` |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                                                                               |   `11.004 kB`<br />`4.188 kB` |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                                                                                 |  `57.552 kB`<br />`16.102 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 |  `105.36 kB`<br />`32.451 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                                                                    |    `11.12 kB`<br />`4.096 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.069 kB`<br />`2.053 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                                                                    |   `20.694 kB`<br />`7.691 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressBar.fixture.js'>ProgressBar</abbr>                                                                        |   `13.578 kB`<br />`5.235 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressField.fixture.js'>ProgressField</abbr>                                                                    |   `22.892 kB`<br />`8.057 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                                                                  |    `16.343 kB`<br />`6.11 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |  `32.247 kB`<br />`10.468 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `14.718 kB`<br />`5.886 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroupField.fixture.js'>RadioGroupField</abbr>                                                                   |   `24.024 kB`<br />`8.796 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |   `23.902 kB`<br />`8.187 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/SelectField.fixture.js'>SelectField</abbr>                                                                          |  `32.162 kB`<br />`10.406 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |   `32.52 kB`<br />`10.357 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/SliderField.fixture.js'>SliderField</abbr>                                                                          |  `41.544 kB`<br />`13.046 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |   `44.933 kB`<br />`12.97 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButtonField.fixture.js'>SpinButtonField</abbr>                                                              |  `53.229 kB`<br />`15.232 kB` |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                                                                 |   `20.445 kB`<br />`6.621 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |  `33.865 kB`<br />`10.735 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/SwitchField.fixture.js'>SwitchField</abbr>                                                                          |  `40.176 kB`<br />`12.514 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                                                                        |   `12.261 kB`<br />`4.819 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                                                                      |    `15.571 kB`<br />`5.23 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |   `26.156 kB`<br />`8.508 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/TextareaField.fixture.js'>TextareaField</abbr>                                                                    |   `35.38 kB`<br />`11.157 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `43.405 kB`<br />`15.166 kB` |
| <samp>react-utilities</samp> <br /> <abbr title='bundle-size/SSRProvider.fixture.js'>SSRProvider</abbr>                                                                       |          `180 B`<br />`159 B` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d8bad5b24268bf1a17081fd39e041f55fc9a2ae6'>d8bad5b24268bf1a17081fd39e041f55fc9a2ae6</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-utilities/src/hooks/useScrollbarWidth.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    if (!targetDocument) {
+      return 0;
+    }
+
+    const outer = document.createElement('div');
+    outer.style.visibility = 'hidden';
+    outer.style.overflow = 'scroll';
+
+    const inner = document.createElement('div');
+    outer.appendChild(inner);
```
Comment (layershifter) [inline packages/react-components/react-utilities/src/hooks/useScrollbarWidth.ts:18]:
```suggestion
    const outer = targetDocument.createElement('div');
    outer.style.visibility = 'hidden';
    outer.style.overflow = 'scroll';

    const inner = targetDocument.createElement('div');
    outer.appendChild(inner);
```

**Review by layershifter — APPROVED** [review]

**Pull request merged**